### ISSUE TITLE
import-shadowing: ignore names like "v2" from imports like "math/rand/v2"

### ISSUE THREAD

**Issue by autarch** [mgechev/revive#1435]
**Describe the bug**

If you have an import like `"math/rand/v2"` and a variable named `v2`, `revive` will flag this as shadowing the import, even though the import ends up named `rand`, not `v2`.

**To Reproduce**

I made a repo that demonstrates the issue at https://github.com/autarch/revive-test

The config just looks like this:

```toml
enableAllRules = true
```

The code in `main.go` is this:

```go
// main is a test package.
package main

import (
	"fmt"
	"math/rand/v2"
)

func main() {
	v2 := "foo"
	fmt.Println(v2)
	fmt.Println(rand.Int())
}
```

**Expected behavior**

I would not expect `revive` to flag any issues with this code. But using the latest revive, I get this:

```
$> revive -version
version 1.11.0
$> revive -config ./revive.toml ./main.go
./main.go:10:2: The name 'v2' shadows an import name
```

**Comment by ccoVeille** [comment]
Good catch. Especially, if we consider that the opposite should be detected

```go
// main is a test package.
package main

import (
	"fmt"
	"math/rand/v2"
)

var _ = rand.Int

func main() {
	rand := "foo"
	fmt.Println(rand)
}
```

**Comment by vbvictor** [comment]
IMO, it's strange to use "bug" together with "nice to have". Given that usually "bug" hurts users right now and "nice to have" doesn't, these tags are conflicting with each other.
This is not a big deal but giving distinct tags could help prioritize development to some extent.

**Comment by alexandear** [comment]
Thanks for the feedback!
This is indeed a minor thing, and it's quite rare for the codebase to use variables like `v1`, `v2`, `v3`, etc., so its impact is limited.

I agree that clearer tagging could help with prioritization - noted for future triage. Removed the `bug` label.
